### DISCUSSION TITLE
invalid byte sequence in UTF-8 (ArgumentError) & 'parse_error!': unmatched end tag at fluentd.con line so so (Fluent::ConfigParseError)

### DISCUSSION THREAD

**Discussion by book-thief** [fluent/fluentd#4338] [Q&A]
Hello

I'm having this error when i download fluentd config  from server and write it to fluentd.con  in ` C:/opt/fluent/etc/fluent/fluentd.conf``

i'm using a powershell script to do it will which is right below

`$CONFFILE = "C:/opt/fluent/etc/fluent/fluentd.conf"	
Write-Output "Downloading config file..."
try {
    $response = Invoke-RestMethod -Uri $CONF_URL -Method Post -Headers $headers -Body $body
	Write-Output $response
	$response | Out-File -FilePath $CONFFILE -Encoding utf8
	Write-Output "Config file downloaded successfully"
} catch {
    Write-Output "Error occurred: $_"
    if ($_.Exception.Response -ne $null) {
        $reader = [System.IO.StreamReader]::new($_.Exception.Response.GetResponseStream())
        $responseBody = $reader.ReadToEnd()
        Write-Output "Response: $responseBody"
    }
}`

and config i get from server is also written below

<source>
  @type windows_eventlog2
  @id windows_eventlog2
  channels ["application", "system", "security", "setup"]
  read_existing_events false
  tag=""
  rate_limit 200
  <parse>
    @type syslog
    message_format auto
  </parse>
  <storage>
    @type local
    persistent true
    path C:\opt\fluent\winlog.json
  </storage>
</source>
<filter **>
  @type record_transformer
  char_encoding ASCII-8BIT:utf-8
  <record>
    time ${time}
    tag ${tag}
  </record>
</filter>
<match **>
  @type http
  endpoint https://.......................................................
  open_timeout 2
  <format>
    @type json
  </format>
  json_array true
</match>

![Screenshot from 2023-11-03 00-24-08](https://github.com/fluent/fluentd/assets/55575715/d8fd60a3-a38c-440a-a2dd-33a5790bcc0d)

as you can see in api call when i get response i encode it to UTF-8 why i stop doing that i get this error and when i encode it  in  UTF-8
Then i get following error

'parse_error!': unmatched end tag at fluentd.con line so so (Fluent::ConfigParseError)

 config is same in  both cases the only difference is encoding and without encodeing
 
 i have attached picture for both error 

![WhatsApp Image 2023-11-03 at 12 18 57 AM](https://github.com/fluent/fluentd/assets/55575715/16c080b7-4b6e-4586-8b4f-80231931b768)

![WhatsApp Image 2023-11-03 at 12 13 13 AM](https://github.com/fluent/fluentd/assets/55575715/0d41f80d-aed5-4d31-9b50-b7787b9dee5f)

**Comment by kenhys** [comment]
Above .conf is not fully formatted well.

Anyway, it seems that `tag=""` raise config error, but not match to your error logs. :thinking: 
It may better to check syntax  `<source>` block.
